File: ruby/reverse_string.rb
require 'pry'
def reverse_string(str)
  reversed = ''
  counter = str.size - 1
  str.each_char do |char|
    # binding.pry
    reversed << str[counter]
    counter -= 1
  end
  reversed
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file

# create counter = 1
# iterate over the word starting from word.length - counter to start with the last letter
# counter + 1

# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  let reversed = ''
  for (let i = 0; i < str.length; i++) {
    reversed = str[i] + reversed
  }
  return reversed
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file

// initialize empty string to hold new word
// itirate over word starting from last letter
// add eache letter of itiration to empty string until beginning of string is reached.


// And a written explanation of your solution

// I initialized a variable "reversed" to an empty string. 
// I started the counter from 1 and looked for the index of str.length minus the counter
// added that letter to the empty string then added 1 to the counter and repeat.
// when i finished the iterations I returned the reversed word.
